src/Sarif.Viewer.VisualStudio/ErrorList/ErrorListService.cs

```diff
@@ -34,139 +37,168 @@ public class ErrorListService
+                        return;
                     }
                 }
-                else
-                {
-                    // The version property wasn't found within the first 100 characters.
-                    // Per the spec, it should appear first in the sarifLog object.
-                    VsShellUtilities.ShowMessageBox(ServiceProvider.GlobalProvider,
-                                                    Resources.VersionPropertyNotFound_DialogTitle,
-                                                    null, // title
-                                                    OLEMSGICON.OLEMSGICON_QUERY,
-                                                    OLEMSGBUTTON.OLEMSGBUTTON_OK,
-                                                    OLEMSGDEFBUTTON.OLEMSGDEFBUTTON_FIRST);
-                    return;
-                }
             }
             else
             {
                 // They're opening a non-SARIF log, so we need to convert it.
                 // Ask if they'd like to save the converted log.
-                MessageDialogCommand response = promptOnLogConversions ? PromptToSaveProcessedLog(Resources.ConvertNonSarifLog_DialogMessage) : MessageDialogCommand.No;
+                MessageDialogCommand response = promptOnLogConversions ?
+                    await PromptToSaveProcessedLogAsync(Resources.ConvertNonSarifLog_DialogMessage).ConfigureAwait(continueOnCapturedContext: false) :
+                    MessageDialogCommand.No;
 
                 if (response == MessageDialogCommand.Cancel)
                 {
                     return;
                 }
 
-                var converter = new ToolFormatConverter();
-                var sb = new StringBuilder();
-
-                using (var input = new MemoryStream(File.ReadAllBytes(filePath)))
-                {
-                    var outputTextWriter = new StringWriter(sb);                
-                    var outputJson = new JsonTextWriter(outputTextWriter);
-                    var output = new ResultLogJsonWriter(outputJson);
-
-                    input.Seek(0, SeekOrigin.Begin);
-                    converter.ConvertToStandardFormat(toolFormat, input, output);
-
-                    // This is serving as a flush mechanism.
-                    output.Dispose();
+                // The converter doesn't have async methods, so spin
+                // up a task to do this.
+                await System.Threading.Tasks.Task.Run(() => {
+                    var sb = new StringBuilder();
+                    using (FileStream fileStream = File.OpenRead(filePath))
+                    {
+                        using (var outputTextWriter = new StringWriter(sb))
+                        using (var outputJson = new JsonTextWriter(outputTextWriter))
+                        using (var output = new ResultLogJsonWriter(outputJson))
+                        {
+                            fileStream.Seek(0, SeekOrigin.Begin);
```

Comment:
Bizarre that you think the stream has moved from start?

Reply:
I thought the same thing (this was from original code). I've removed it.

---
In reply to: [494562325](https://github.com/microsoft/sarif-visualstudio-extension/pull/202#discussion_r494562325) [](ancestors = 494562325)